plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -385,8 +397,10 @@ public void updatePartitionStatistics(HiveIdentity identity, Table table, String
 
         try {
             PartitionInput partitionInput = GlueInputConverter.convertPartition(partition);
-            partitionInput.setParameters(updateStatisticsParameters(partition.getParameters(), updatedStatistics.getBasicStatistics()));
-            columnStatisticsProvider.updatePartitionStatistics(partitionInput, updatedStatistics.getColumnStatistics());
+            final Map<String, String> updateStatisticsParameters = updateStatisticsParameters(partition.getParameters(), updatedStatistics.getBasicStatistics());
+            partitionInput.setParameters(updateStatisticsParameters);
+            partition = Partition.builder(partition).setParameters(updateStatisticsParameters).build();
+            columnStatisticsProvider.updatePartitionStatistics(partition, updatedStatistics.getColumnStatistics());
```

Comment:
please move call to `columnStatisticsProvider.updatePartitionStatistics` below `glueClient.updatePartition` we update the basic stats first.